### PR TITLE
Now we reject promises with an Error that has a stack trace

### PR DESCRIPTION
**Pull request by digeff**
It'd be useful to have a full stack trace when a message fails

### CODE REVIEW

File: lib/noice-json-rpc.js [hunk 1]
```diff
@@ -49,7 +62,7 @@ class Client extends events_1.EventEmitter {
                     promise.resolve(message.result);
                 }
                 else if (message.error) {
-                    promise.reject(message.error);
+                    promise.reject(new MessageError(message.error.code, message.error.message, message.error.data));
```
Comment (nojvek) [inline lib/noice-json-rpc.js]: I'd prefer 
```
promise.reject(new MessageError(message.error))
```

File: src/noice-json-rpc.ts [hunk 2]
```diff
@@ -27,6 +27,25 @@ export interface ClientOpts extends LogOpts {
 export interface ServerOpts extends LogOpts {
 }
 
+export class MessageError extends Error implements JsonRpc2.Error {
```
Comment (nojvek) [inline src/noice-json-rpc.ts:30]: perhaps just a simpler class definition  ? wdyt ?
```
export class MessageError extends Error implements JsonRpc2.Error {
  constructor(error: JSONRpc2.Error) {
    super(error.message)
    this.code = error.code || 0;
    this.data = error.data || null;
  }
}

```

File: lib/noice-json-rpc.js (hunk 1 above)
Comment (digeff) [reply lib/noice-json-rpc.js]: Done

File: src/noice-json-rpc.ts (hunk 2 above)
Comment (digeff) [reply src/noice-json-rpc.ts:30]: Done

**Review by nojvek — APPROVED** [review]

**Comment by nojvek** [comment]
Thanks digeff

**Comment by digeff** [comment]
Thanks @nojvek. When can we expect a new release of the npm package with this change?

**Comment by nojvek** [comment]
+ noice-json-rpc@1.1.2

Sorry, I'm on vacation at the moment in Australia, so my computer time is spotty.

**Comment by digeff** [comment]
Thank you very much :)

**Pull request merged**